### PR TITLE
given inference model path, download model automatically

### PR DESCRIPTION
**Pull request by XYZ-916**
given inference model path, download model automatically:

1. add download.py file;
2. modify the method in pipeline.py to get model dir;
3. change model dir in infer_cfg_pphuman.yml to url.

### CODE REVIEW

File: deploy/pipeline/pipeline.py [hunk 1]
```diff
@@ -292,17 +294,48 @@ def __init__(self, args, cfg, is_video=True, multi_camera=False):
         self.file_name = None
         self.collector = DataCollector()
 
+        # auto download inference model
+        model_dir_dict = {}
+        for key in self.cfg.keys():
```
Comment (zhiboniu) [inline deploy/pipeline/pipeline.py]: 这一段封成函数调用吧

File: deploy/pipeline/config/infer_cfg_pphuman.yml [hunk 2]
```diff
@@ -5,28 +5,28 @@ visual: True
...
   enable: False
 
 KPT:
-  model_dir: output_inference/dark_hrnet_w32_256x192/
+  model_dir: https://bj.bcebos.com/v1/paddledet/models/pipeline/dark_hrnet_w32_256x192.zip
   batch_size: 8
 
 ATTR:
-  model_dir: output_inference/PPLCNet_x1_0_person_attribute_945_infer/
+  model_dir:  https://bj.bcebos.com/v1/paddledet/models/pipeline/PPLCNet_x1_0_person_attribute_945_infer.tar
```
Comment (jerrywgz) [inline deploy/pipeline/config/infer_cfg_pphuman.yml]: 这里的格式统一成zip吧

File: deploy/pipeline/pipeline.py [hunk 3]
```diff
@@ -292,17 +294,48 @@ def __init__(self, args, cfg, is_video=True, multi_camera=False):
...
+                    or "enable" not in self.cfg[key].keys()):
+
+                if "model_dir" in self.cfg[key].keys():
+                    model_dir = self.cfg[key]["model_dir"]
+                    downloaded_model_dir = auto_download_model(model_dir)
+                    if downloaded_model_dir:
+                        model_dir = downloaded_model_dir
+                    model_dir_dict[key] = model_dir
+                    print(key, " model dir:", model_dir)
+                elif "VEHICLE_PLATE" == key:
```
Comment (jerrywgz) [inline deploy/pipeline/pipeline.py]: key == "VEHICLE_PLATE"

File: deploy/pipeline/pipeline.py [hunk 4]
```diff
@@ -292,17 +294,48 @@ def __init__(self, args, cfg, is_video=True, multi_camera=False):
         self.file_name = None
         self.collector = DataCollector()
 
+        # auto download inference model
```
Comment (jerrywgz) [inline deploy/pipeline/pipeline.py:336]: 需要更新使用文档

File: deploy/pipeline/config/infer_cfg_pphuman.yml (hunk 2 above)
Comment (XYZ-916) [reply deploy/pipeline/config/infer_cfg_pphuman.yml]: 这里可能需要对应的模型修改成zip压缩包后重新上传。

File: deploy/pipeline/pipeline.py (hunk 1 above)
Comment (XYZ-916) [reply deploy/pipeline/pipeline.py]: ok

**Review by zhiboniu — DISMISSED** [review]

**Review by jerrywgz — APPROVED** [review]

**Review by zhiboniu — APPROVED** [review]

**Pull request merged**